File: Service/MediCare/RMA.Service.MediCare.Contracts/Entities/Medical/MedicalInvoicePaymentAllocation.cs
using RMA.Common.Entities;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.MediCare.Contracts.Entities.Medical
{
    public class MedicalInvoicePaymentAllocation : AuditDetails
    {
        public int AllocationId { get; set; } // AllocationId (Primary key)
        public int PayeeId { get; set; } // PayeeId
        public PaymentAllocationStatusEnum PaymentAllocationStatus { get; set; } // AllocationStatusId
        public int? MedicalInvoiceId { get; set; } // MedicalInvoiceId
        public int? TebaInvoiceId { get; set; } // TebaInvoiceId
        public int? DaysOffInvoiceId { get; set; } // DaysOffInvoiceId
        public int? PdAwardId { get; set; } // PDAwardId
        public decimal AssessedAmount { get; set; } // Amount
        public decimal AssessedVat { get; set; } // Amount
        public PaymentTypeEnum? PaymentType { get; set; } // PaymentTypeId
    }
}

File: Web/RMA-Web-Shell/projects/medicare/src/app/medical-invoice-manager/models/medical-invoice-payment-allocation.ts
import { PaymentAllocationStatusEnum } from "projects/fincare/src/app/shared/enum/payment-allocation-status-enum";
import { PaymentTypeEnum } from "projects/shared-models-lib/src/lib/enums/payment-type-enum";

export class MedicalInvoicePaymentAllocation {
    allocationId: number;
    payeeId: number;
    paymentAllocationStatus: PaymentAllocationStatusEnum;
    medicalInvoiceId: number | null;
    daysOffInvoiceId: number | null;
    pdAwardId: number | null;
    assessedAmount: number;
    assessedVat: number;
    isActive: boolean;
    isDeleted: boolean;
    createdBy: string;
    createdDate: string;
    modifiedBy: string;
    modifiedDate: string;
    paymentType: PaymentTypeEnum | null;
}
